extensions/vscode/src/eventErrors.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2024 by Posit Software, PBC.
+
+import { EventStreamMessage } from "./api/types/events";
+import { ErrorCode } from "./utils/errorTypes";
+
+export interface EventStreamMessageErrorCoded<T = Record<string, string>>
+  extends EventStreamMessage<T> {
+  errCode: ErrorCode;
+}
+
+export function isCodedEventErrorMessage(
+  msg: EventStreamMessage,
+): msg is EventStreamMessageErrorCoded {
+  return msg.errCode !== undefined;
+}
+
+type deploymentEvtErr = {
+  dashboardUrl: string;
+  localId: string;
+  error: string;
+};
+
+export const isEvtErrTargetNotFound = (
+  emsg: EventStreamMessageErrorCoded,
+): emsg is EventStreamMessageErrorCoded<deploymentEvtErr> => {
+  return emsg.errCode === "deploymentTargetNotFound";
+};
+
+export const isEvtErrTargetForbidden = (
+  emsg: EventStreamMessageErrorCoded,
+): emsg is EventStreamMessageErrorCoded<deploymentEvtErr> => {
+  return emsg.errCode === "deploymentTargetIsForbidden";
+};
+
+export const handleEventCodedError = (
+  emsg: EventStreamMessageErrorCoded,
+): string => {
+  if (isEvtErrTargetNotFound(emsg)) {
+    return `Content at ${emsg.data.dashboardUrl} could not be found. Please, verify the content "id" is accurate.`;
```

Comment:
I don't think this is an error that most users will be able to fix on their own. 

This is because the `id` is inside of the Content Record under `.posit/publish/deployments` which we never prompt the user to open, and is a little tough to find the correct deployment for your configuration if you have multiple / don't know what to look for.

This is a much better error that what we have now, so perhaps what I'm suggesting should be a follow-up, but we should have a way for users to solve this issue.

Reply:
I think this provides enough info for now, considering it might be very likely to happen when associating content, the context of the interaction feels appropriate. Totally agree that there's room for improvement but user feedback may provide a better direction.